### PR TITLE
Implement --pyk option for krun as well

### PR DESCRIPTION
**Pull request by ehildenb**
This makes it so that all major functionality of the `kevm` runner script can be diverted through the `kevm_pyk` runner instead.

- Flag `--pyk` is now read by `kevm run` and `kevm interpret` to divert control to the `kevm_pyk` script.
- Rename Makefile variable `TEST_OPTIONS => KEVM_OPTS`, and thread it through the Makefile.
- Add `Makefile` variables `KRUN_OPTS`, `KSEARCH_OPTS` and add in the appropriate places.
- The `kevm_pyk` script supports the needed options to run `make test-conformance KEVM_OPTS=--pyk`.
- Turns on `--emit-json` by default for `kevm_pyk kompile`, but allows turning it off.
- Adds another directory to `make venv-clean` for Ubuntu Jammy.
- Hooks up `kevm_pyk --profile ...` to the appropriate places so that sub-processes are profiled on runs.

Here is some sample output produced running `make test-kevm-pyk` on my machine:

```
8-13 20:42:30,486 pyk.ktool.krun - Running: krun --definition /home/dev/src/evm-semantics/.build/usr/lib/kevm/llvm /tmp/tmp.fysbfVW76y --output json --term --parser cat --no-expand-macros --output kore
8-13 20:42:30,554 pyk.ktool.krun - Timing [0.068]: krun --definition /home/dev/src/evm-semantics/.build/usr/lib/kevm/llvm /tmp/tmp.fysbfVW76y --output json --term --parser cat --no-expand-macros --output kore
8-13 20:42:30,554 pyk.ktool.krun - Completed: krun --definition /home/dev/src/evm-semantics/.build/usr/lib/kevm/llvm /tmp/tmp.fysbfVW76y --output json --term --parser cat --no-expand-macros --output kore
1.310s 0m0.092s python3 -m kevm_pyk run --definition /home/dev/src/evm-semantics/.build/usr/lib/kevm/llvm /tmp/tmp.fysbfVW76y --schedule LONDON --mode NORMAL --chainid 1 --term --parser cat --no-expand-macros --output kore --profile --definition /home/dev/src/evm-semantics/
```

So we can see here that the actual call to `krun ...` bash script takes 0.068s, and the overall call to `kevm_pyk run ...` takes 1.310s. So we introduce about 1.3s of overhead by running through pyk. This is lower than the original 4s because of this: https://github.com/runtimeverification/k/pull/2800.

### CODE REVIEW

**Comment by ehildenb** [comment]
I have compared performance of `make test-conformance -j20` and `make test-conformance -j20 TEST_OPTIONS=--pyk` here. The breakdown is:

```
....... Many thousands lines of data ......
 GeneralStateTests/stSStoreTest/sstore_0to0_d2g1v0.json                                                                       | 4.78              | 0.08                       | 59.75                                
 GeneralStateTests/stSStoreTest/sstore_0to0toX_d2g0v0.json                                                                    | 4.78              | 0.08                       | 59.75                                
 GeneralStateTests/stSStoreTest/sstore_0to0toX_d3g1v0.json                                                                    | 4.78              | 0.08                       | 59.75                                
 GeneralStateTests/stSStoreTest/sstore_0toX_d5g1v0.json                                                                       | 4.78              | 0.08                       | 59.75                                
 GeneralStateTests/stSStoreTest/sstore_0to0to0_d0g1v0.json                                                                    | 4.79              | 0.08                       | 59.875                               
 GeneralStateTests/stSStoreTest/sstore_0to0to0_d2g1v0.json                                                                    | 4.79              | 0.08                       | 59.875                               
 GeneralStateTests/stSStoreTest/sstore_0to0toX_d3g0v0.json                                                                    | 4.79              | 0.08                       | 59.875                               
 GeneralStateTests/stSStoreTest/sstore_0toX_d3g1v0.json                                                                       | 4.81              | 0.08                       | 60.12499999999999                    
 GeneralStateTests/stSStoreTest/sstore_0to0to0_d9g1v0.json                                                                    | 4.82              | 0.08                       | 60.25                                
 GeneralStateTests/stSStoreTest/sstore_0to0to0_d4g0v0.json                                                                    | 4.85              | 0.08                       | 60.62499999999999                    
 GeneralStateTests/stSStoreTest/sstore_0toX_d7g0v0.json                                                                       | 4.85              | 0.08                       | 60.62499999999999                    
 GeneralStateTests/stSStoreTest/sstore_0toX_d2g0v0.json                                                                       | 4.86              | 0.08                       | 60.75                                
 GeneralStateTests/stSStoreTest/sstore_0to0_d0g0v0.json                                                                       | 4.87              | 0.08                       | 60.875                               
 GeneralStateTests/stSStoreTest/sstore_0to0to0_d1g0v0.json                                                                    | 4.87              | 0.08                       | 60.875                               
 GeneralStateTests/stSStoreTest/sstore_0to0toX_d2g1v0.json                                                                    | 4.87              | 0.08                       | 60.875                               
 GeneralStateTests/stSStoreTest/sstore_0to0toX_d6g0v0.json                                                                    | 4.87              | 0.08                       | 60.875                               
 GeneralStateTests/stSStoreTest/sstore_0to0toX_d8g1v0.json                                                                    | 4.87              | 0.08                       | 60.875                               
 GeneralStateTests/stSStoreTest/sstore_0toXto0_d4g0v0.json                                                                    | 4.87              | 0.08                       | 60.875                               
 GeneralStateTests/stSStoreTest/sstore_0toX_d8g1v0.json                                                                       | 4.88              | 0.08                       | 61.0                                 
 GeneralStateTests/stEIP3607/transactionCollidingWithNonEmptyAccount_send.json                                                | 4.35              | 0.07                       | 62.14285714285713                    
 GeneralStateTests/stSStoreTest/sstore_0toX_d6g1v0.json                                                                       | 4.98              | 0.08                       | 62.25000000000001                    
 GeneralStateTests/stStaticCall/static_CallEcrecover0_0input_d7g0v0.json                                                      | 4.36              | 0.07                       | 62.285714285714285                   
 GeneralStateTests/stSStoreTest/sstore_0to0toX_d6g1v0.json                                                                    | 5.0               | 0.08                       | 62.5                                 
 VMTests/vmLogTest/log3_logMemStartTooHigh.json                                                                               | 4.46              | 0.07                       | 63.71428571428571                    
 GeneralStateTests/stInitCodeTest/OutOfGasContractCreation_d0g0v0.json                                                        | 4.46              | 0.07                       | 63.71428571428571                    
 GeneralStateTests/stSStoreTest/sstore_0to0_d0g1v0.json                                                                       | 5.1               | 0.08                       | 63.74999999999999                    
 TOTAL                                                                                                                        | 61770.37999999977 | 2277.40999999916           | 27.123082800208376      
```

So we can see that we introduce about ~4.5s overhead into each execution, which overall is a massive slowdown, using teh `--pyk` option. We need to optimize this, but at least we are able to measure it now!

File: Makefile [hunk 1]
```diff
@@ -646,18 +646,20 @@ test-failure: $(failure_tests:=.run-expected)
...
+                  tests/foundry/foundry.k.check                                                                \
+                  tests/specs/bihu/functional-spec.k.prove                                                     \
+                  tests/specs/examples/empty-bin-runtime.k                                                     \
+                  tests/specs/examples/erc20-bin-runtime.k                                                     \
                   tests/specs/examples/erc721-bin-runtime.k
 
 test-kevm-pyk: KPROVE_OPTS  += --pyk --verbose
 test-kevm-pyk: KOMPILE_OPTS += --pyk --verbose
 test-kevm-pyk: KAST_OPTS += --pyk --verbose
+test-kevm-pyk: TEST_OPTIONS += --pyk --verbose
```
Comment (radumereuta) [inline Makefile:662]: Why is this called TEST_OPTIONS? The rest have tool names, except for this one.

**Review by radumereuta — COMMENTED** [review]
I think I understand what you are trying to do, but I will still like it if it's approved by someone with more python experience.
I had just one question about naming a variable.

File: Makefile (hunk 1 above)
Comment (ehildenb) [reply Makefile]: Good call. I renamed `TEST_OPTIONS` to `KEVM_OPTIONS`, and now that's the only variable that needs to be set. I've also threaded that variable through ot all the places it should be present, and added variables `KRUN_OPTS`, `KAST_OPTS`, and `KSEARCH_OPTS`.

**Review by tothtamas28 — APPROVED** [review]

**Pull request merged**